pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -186,17 +213,13 @@ func (c *Controller) Waypoint(scope model.WaypointScope) sets.Set[netip.Addr] {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	res := sets.New[netip.Addr]()
-	for ip := range a.waypoints[scope] {
-		res.Insert(netip.MustParseAddr(ip))
-	}
-	if len(res) > 0 {
-		// Explicit has precedence
-		return res
+	if ip, f := a.waypoints[scope]; f {
```

Comment:
Waypoint() now always returns 0 or 1 so probably don't need a Set

Reply:
I wasn't sure how to handle that in the `pilot/pkg/serviceregistry/aggregate/controller.go.Waypoint()` since calls all the registries, appending the results.

Reply:
Well the proto only allows 1 now so we may need to change that behavior. Probably want the local-cluster's VIP

Reply:
I think this should be changed - but will address in follow-up since this PR is getting big